cpp/src/arrow/testing/fixed_width_test_util.h

```diff
@@ -17,16 +17,6 @@
 
 #pragma once
 
-#include <cstddef>
-#include <cstdint>
-#include <memory>
-#include <vector>
-
-#include "arrow/array/builder_primitive.h"
-#include "arrow/builder.h"
-#include "arrow/type.h"
-#include "arrow/util/checked_cast.h"
-
```

Comment:
I meant reducing the code in the header. We should keep the necessary includes.

The class should only declare the public function members here and all the private members and template code can be defined in the `.cc` file (to be created) outside of the class.

See how `chunk_resolver.h/cc` does it reducing the bloat in the header: https://github.com/apache/arrow/blob/51689a040cbe3dee8702cd899a33fa62e0616bf1/cpp/src/arrow/chunk_resolver.cc#L44-L45

Reply:
Ah, so I need to organize the code according to C++ conventions. I should include only the declarations in the .h file and put the implementations in the .cc file. 

I will refer to the code you shared, organize it accordingly, and then upload it.

Thank you for your guidance.